docs/README.md

```diff
@@ -0,0 +1,358 @@
+                struct am_trace* t)
+    {
+      return trace_changed_per_ecoll_array(l, t);
+    }
+
+    static int renderer_changed_<type>(struct am_timeline_render_layer* l,
+                   struct am_timeline_renderer* r)
+    {
+      if(r->trace)
+        return trace_changed_<type>(l, r->trace);
+      else
+        return 0;
+    }
+
+    static size_t calculate_index_<type>(
+      struct am_timeline_interval_layer* l,
+      void* arg)
+    {
+      struct am_<group>_<type>* m = arg;
+
+      // This value determines the colour in the GUI
+      return 0;
+    }
+
+    struct am_timeline_render_layer_type*
+    am_timeline_<group>_<type>_layer_instantiate_type(void)
+    {
+      struct am_timeline_render_layer_type* t;
+
+      t = am_timeline_interval_layer_instantiate_type_index_fun(
+        "<group>::<type>",
+        "am::<group>::<type>",
+        sizeof(struct am_<group>_<type>),
+        offsetof(struct am_<group>_<type>, interval),
+        calculate_index_<type>);
+
+      t->trace_changed = trace_changed_<type>;
+      t->renderer_changed = renderer_changed_<type>;
+
+      return t;
+    }
+    ```
+
+    For discrete events use the following code instead:
+
+    ```c
+    struct am_timeline_render_layer_type*
+    am_timeline_<group>_<type>_acquire_layer_instantiate_type(void)
+    {
+      struct am_timeline_render_layer_type* t;
+
+      t = am_timeline_discrete_layer_instantiate_type_index_fun(
+        "<group>::<type>",
+        "am::<group>::<type>",
+        sizeof(struct am_<group>_<type>),
+        offsetof(struct am_<group>_<type>, timestamp),
+        calculate_index_<type>,
+        render_event);
+
+      t->trace_changed = trace_changed_<type>;
+      t->renderer_changed = renderer_changed_<type>;
+
+      return t;
+    }
+    ```
+
+    Where the `render_event` function can be freely implemented.
+
+6. Add timeline type in `libaftermath-render/src/timeline/common_layers.c` by adding
+`am_timeline_<group>_<type>_layer_instantiate_type` to `(*inst_functions[])(void)`.
+
+# Using Aftermath GUI
+
+## Adding new type to GUI
+
+### Extending existing profile
+
+Steps below explain how to add new type to one of the existing profiles
+in Aftermath:
+
+1. Add new icon to `aftermath/share/icons/`.
+
+2. Add path to the new icon to `ICONS` in `aftermath/Makefile.am` as
+`share/icons/draw_<group>_<type>.svg \`.
+
+3. Add new button to `aftermath/share/profiles/<profile>/interface.amgui.in` as for example:
+
+    ```
+    amgui_toolbar_togglebutton {
+                  tooltip: "Some short description",
+                  icon: "@ICON_BASEDIR@/draw_<group>_<type>.svg",
+                  checked: 0u64,
+                  id: "toolbutton_draw_<group>_<type>"
+    ```
+
+    And add the type to list of layers:
+
+    ```
+    amgui_timeline {
+      id: "tl1",
+      layers: [..., "<group>_<type>", ...]
+    ```
+
+4. Update DFG diagram in `aftermath/share/profiles/<profile>/graph.dfg` with
+nodes of the new type.
+
```

Comment:
Suggestion: Update the data-flow graph in ... with nodes and connections of the new type.